main.go

```diff
@@ -458,7 +457,51 @@ func setupDatabase(cctx *cli.Context) (*gorm.DB, error) {
 	}
 
 	db.AutoMigrate(&Content{})
-	db.AutoMigrate(&Object{})
+
+	// Check if we need to migrate cid to hash.
+	if db.Migrator().HasColumn(&Object{}, "hash") {
+		db.AutoMigrate(&Object{})
+	} else {
+		// Migration from cid to hash is needed
+		fmt.Println("migrating object cids to multihashes")
+		type cidrow struct {
+			ID  uint
+			Cid util.DbCID
+		}
+		rows, err := db.Raw("select id, cid from objects").Rows()
+		if err != nil {
+			return nil, err
+		}
+		defer rows.Close()
+
+		if err := db.Transaction(func(tx *gorm.DB) error {
+			if err := tx.Migrator().AddColumn(&Object{}, "hash"); err != nil {
+				return err
+			}
+			count := 0
+			for rows.Next() {
```

Comment:
Any idea how long this would take on 100 million rows? My worry is that it would take hours, and estuary being offline for that long isnt great.

Reply:
Will try and benchmark some different sizes to get an estimate. Informally it's about a second for 10k rows in a docker container on my machine.

Reply:
We can also do it as a multistage. Add the column and write to both cid+hash during every insert. In background update all the hashes. We would have to rewrite where clauses to use cid OR hash. 

Then some time in the future we remove the cid column. We would have to be careful not to break people who have not deployed and run the first stage beforehand.

Reply:
yeah, so at that speed thats over two hours for the full migration, which means the application is offline for that long, which would be pretty bad.

That said, maybe we could add some parallelism? I think I would be okay with under an hour of downtime if you think we could hit that. Then we could just schedule a time for the migration to happen and do it then.

Reply:
Also, I think @lanzafame has backups of the database laying around, we could try it out on a replica